platform/src/main/java/org/stellar/anchor/platform/config/PropertyEventConfig.java

```diff
@@ -31,15 +26,54 @@ public void validate(Object target, Errors errors) {
       return;
     }
 
-    ValidationUtils.rejectIfEmptyOrWhitespace(errors, "publisherType", "");
+    // Validate publisher type
+    ValidationUtils.rejectIfEmptyOrWhitespace(errors, "publisher.type", "");
+
+    PublisherConfig publisherConfig = config.getPublisher();
+    String publisherType = publisherConfig.getType();
+    switch (publisherType) {
+      case "msk":
+        ValidationUtils.rejectIfEmptyOrWhitespace(
+            errors,
+            "publisher.msk.useIAM",
+            "empty-msk-use-iam",
+            "use_IAM must be defined for MSK publisher");
+        // continue to the kafka case. DO NOT break
+      case "kafka":
+        ValidationUtils.rejectIfEmptyOrWhitespace(
+            errors, "publisher.kafka.bootstrapServers", "empty-bootstrapServer");
+        ValidationUtils.rejectIfEmptyOrWhitespace(
+            errors,
+            "publisher.kafka.retries",
+            "empty-retries",
+            "retries must be set for KAFKA/MSK publisher");
+        ValidationUtils.rejectIfEmptyOrWhitespace(
+            errors,
+            "publisher.kafka.batchSize",
```

Comment:
👀 🤔  You're using camelCase here but the yaml actually uses snake case. Shouldn't it be `batch_size`?  The same goes for other configuration validations in this file.

I've tested locally and it seems that camelCase validation is ignored, while the snake_case one does the intended check. Can you test it on your end too?

Reply:
I tested it locally on my end and it works.

Reply:
I may have made a mistake then. It'd be good to add a test for this like we do in these files https://github.com/stellar/java-stellar-anchor-sdk/tree/ee620546fbb2c3e800f18a649d613d09839272d6/platform/src/test/kotlin/org/stellar/anchor/platform/config.